integration/build/build_test.go

```diff
@@ -0,0 +1,129 @@
+package build
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/pkg/jsonmessage"
+
+	"github.com/docker/docker/api/types/filters"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/util/request"
+)
+
+func TestBuildWithRemoveAndForceRemove(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name                           string
+		dockerfile                     string
+		numberOfIntermediateContainers int
+		rm                             bool
+		forceRm                        bool
+	}{
+		{
+			name: "successfull-false-false",
```

Comment:
These names could be a lot more descriptive. The dashes can also be spaces, `testing.Run()` will take care of adding the dashes.  

* "successful build with no removal"
* "successful build with remove"
* "successful build with remove and force remove"
* "failed build with no removal"
...